neard/src/runtime.rs

```diff
@@ -128,6 +129,7 @@ pub struct NightshadeRuntime {
     genesis_runtime_config: Arc<RuntimeConfig>,
 
     store: Arc<Store>,
+    in_memory_contracts: InMemoryContracts,
```

Comment:
why don't we make it per shard to reduce potentially lock contention?

Reply:
I'd probably suggest to YAGNI that, unless the code is proven guilty for a perf regression. With per-shard, we'll be using N times more memory storing the same bytes. I think a more promising optimization avenue here (should we need it) is to precompile stuff at stratup, and then share a read-only cache.

Reply:
Yeah, make it per shard sounds good

Reply:
Github didn't show @matklad 's comment when i comment previously. I thinks per shard mem is better. We don't try to tell whether it's a same bytes before cache it (it's cached based on account name, not contract bytes), so won't spend more memory.

Reply:
@matklad yeah there will be no repetition yet unless we introduce contracts that are globally accessible/

Reply:
Ack! might make sense to shard this from the start then, if this doesn't make the code more complex.